code/game/objects/items/weapons/improvised_components.dm

```diff
@@ -68,3 +68,34 @@
 		qdel(src)
 		user.put_in_hands(finished)
 	update_icon(user)
+
+/obj/item/woodcirclet
+	name = "wood circlet"
+	desc = "A small wood circlet for making a flower crown."
+	icon = 'icons/obj/buildingobject.dmi'
+	icon_state = "woodcirclet"
+	w_class = ITEMSIZE_SMALL
+
+/obj/item/woodcirclet/attackby(obj/item/W as obj, mob/user as mob)
+	var/obj/item/complete
+	if(istype(W,/obj/item/seeds/poppyseed))
+		user << "You attach the poppy to the circlet and create a beautiful flower crown."
+		complete = new /obj/item/clothing/head/poppy_crown(get_turf(user))
+		qdel(W)

```

Comment:
Before calling qdel(W), it would be best to call user.drop_from_inventory(W) before deleting it, or some weirdness may occur.